lib/MqttClient.js

```diff
@@ -269,23 +261,49 @@ class MqttClient {
 
     /**
      * @private
-     * @param {import("./dtos/MapDTO")} mapDTO
+     * @param {import("./entities/map/ValetudoMap")} map
      */
-    updateMapDataTopic(mapDTO) {
-        if (this.client && this.client.connected === true && mapDTO && mapDTO.parsedData) {
-            this.client.publish(this.topics.map_data, JSON.stringify(mapDTO.parsedData), {retain: true, qos:this.qos});
+    updateMapDataTopic(map) {
+        if (this.client && this.client.connected === true && map) {
+            this.client.publish(this.topics.map_data, JSON.stringify(map), {retain: true, qos:this.qos});
         }
     }
 
     /**
      * @private
-     * @param {import("./miio/Status")} statusData
+     * @param {import("./entities/state/RobotState")} robotState
      */
-    updateAttributesTopicOnEvent(statusData) {
-        this.last_ha_state = HA_STATE_MAPPINGS[statusData.state];
-        this.last_state = statusData.state;
-        this.last_cleaned_area = statusData.clean_area ? parseFloat((statusData.clean_area / 1000000).toFixed(2)) : 0;
-        this.last_cleaning_time = statusData.clean_time ? parseFloat((statusData.clean_time / 60).toFixed(1)) : 0;
+    updateAttributesTopicOnEvent(robotState) {
+        const StatusStateAttribute = robotState.getFirstMatchingAttributeByConstructor(attributes.StatusStateAttribute);
+        const areaStatistic = robotState.getFirstMatchingAttribute({
+            attributeClass: attributes.LatestCleanupStatisticsAttribute.name,
+            attributeType: attributes.LatestCleanupStatisticsAttribute.TYPE.AREA
+        });
+        const durationStatistic = robotState.getFirstMatchingAttribute({
+            attributeClass: attributes.LatestCleanupStatisticsAttribute.name,
+            attributeType: attributes.LatestCleanupStatisticsAttribute.TYPE.DURATION
+        });
+
+        if (StatusStateAttribute) {
+            this.last_ha_state = HA_STATE_MAPPINGS[StatusStateAttribute.value];
+            this.last_state = StatusStateAttribute.value;
+
+            if (StatusStateAttribute.flag !== attributes.StatusStateAttribute.FLAG.NONE) {
+                this.last_state += " " + StatusStateAttribute.flag;
+            }
+        }
+
+        if (areaStatistic) {
+            this.last_cleaned_area = parseFloat((areaStatistic.value / 10000).toFixed(2));
```

Comment:
We should use the [round](https://github.com/Hypfer/Valetudo/blob/bc29f4eed26516682bb91e3771f3346902680e5b/lib/MqttClient.js#L55) function